model/textparse/interface.go

```diff
@@ -91,7 +91,7 @@ func New(b []byte, contentType string, parseClassicHistograms bool, st *labels.S
 	}
 	switch mediaType {
 	case "application/openmetrics-text":
-		return NewOpenMetricsParser(b, st), nil
+		return NewOpenMetricsParser(b, st, WithOMParserCTSeriesSkipped()), nil
```

Comment:
Similarly to us trying to not break `NewOpenMetricsParser` users by adding opt-in setting, ideally we could do the same for `New`, and let scrape loop explicitly skip OMs.

If we want keep this opt-in in `NewOpenMetricsParser` and `hardcoded` to skip in `New`, we might as well make it hardcoded in `NewOpenMetricsParser`.

To sum up, I would add another bool next to `parseClassicHistograms` as `skipOMSeries` and let `New` callers decide.

Reply:
ah yes that makes sense, it can then be enabled based off `sl.enableCTZeroIngestion`

pushed changes